src/openrct2/interface/Viewport.cpp

```diff
@@ -1047,11 +1043,11 @@ CoordsXY screen_pos_to_map_pos(int16_t x, int16_t y, int32_t* direction)
     return mapCoords;
 }
 
-LocationXY16 screen_coord_to_viewport_coord(rct_viewport* viewport, uint16_t x, uint16_t y)
+LocationXY16 screen_coord_to_viewport_coord(rct_viewport* viewport, ScreenCoordsXY screenCoords)
```

Comment:
A viewport coord is also a screen coord just adjusted for zoom. So in a future refactor this could be refactored to return a ScreenCoord struct.

Reply:
I saved it to do it next.